include/ignition/gazebo/World.hh

```diff
@@ -0,0 +1,192 @@
+      /// \param[in] _world World to copy.
+      /// \return Reference to this.
+      public: World &operator=(const World &_world);
+
+      /// \brief Destructor
+      public: virtual ~World();
+
+      /// \brief Get the entity which this World is related to.
+      /// \return World entity.
+      public: gazebo::Entity Entity() const;
+
+      /// \brief Check whether this world correctly refers to an entity that
+      /// has a components::World.
+      /// \param[in] _ecm Entity-component manager.
+      /// \return True if it's a valid world in the manager.
+      public: bool Valid(const EntityComponentManager &_ecm) const;
+
+      /// \brief Get the link's unscoped name.
+      /// \param[in] _ecm Entity-component manager.
+      /// \return Link's name or nullopt if the entity does not have a
+      /// components::Name component
+      public: std::optional<std::string> Name(
+          const EntityComponentManager &_ecm) const;
+
+      /// \brief Get the gravity in m/s^2.
+      /// \param[in] _ecm Entity-component manager.
+      /// \return Gravity vector or nullopt if the entity does not
+      /// have a components::Gravity component.
+      public: std::optional<math::Vector3d> Gravity(
+          const EntityComponentManager &_ecm) const;
+
+      /// \brief Get the magnetic field in T.
+      /// \param[in] _ecm Entity-component manager.
+      /// \return Magnetic field vector or nullopt if the entity does not
+      /// have a components::MagneticField component.
+      public: std::optional<math::Vector3d> MagneticField(
+          const EntityComponentManager &_ecm) const;
+
+      /// \brief Get atmosphere information.
+      /// \param[in] _ecm Entity-component manager.
+      /// \return Magnetic field vector or nullopt if the entity does not
+      /// have a components::Atmosphere component.
+      public: std::optional<sdf::Atmosphere> Atmosphere(
+          const EntityComponentManager &_ecm) const;
+
+      /// \brief Get the ID of a light entity which is an immediate child of
+      /// this world.
+      /// \param[in] _ecm Entity-component manager.
+      /// \param[in] _name Light name.
+      /// \return Light entity.
+      /// \todo(anyone) Make const
+      public: gazebo::Entity LightByName(const EntityComponentManager &_ecm,
+          const std::string &_name);
+
+      /// \brief Get the ID of a actor entity which is an immediate child of
+      /// this world.
+      /// \param[in] _ecm Entity-component manager.
+      /// \param[in] _name Actor name.
+      /// \return Actor entity.
+      /// \todo(anyone) Make const
+      public: gazebo::Entity ActorByName(const EntityComponentManager &_ecm,
+          const std::string &_name);
+
+      /// \brief Get the ID of a model entity which is an immediate child of
+      /// this world.
+      /// \param[in] _ecm Entity-component manager.
+      /// \param[in] _name Model name.
+      /// \return Model entity.
+      /// \todo(anyone) Make const
```

Comment:
I just took a quick look and couldn't see what was keeping this from being const?

Reply:
I just set it to `const` locally, and it didn't seem to have an issue building, maybe worth going ahead and doing?

Reply:
Nice catch, that was carelessly copied from

https://github.com/ignitionrobotics/ign-gazebo/blob/0d16dc96ec0544440a082f37d2cfa72a97dcbc9f/include/ignition/gazebo/Model.hh#L129-L131

Fixed in 6a8a6d0